examples/credential-registry/src/lib.rs

```diff
@@ -636,13 +622,12 @@ pub struct CredentialInfo {
     metadata_url:     MetadataUrl,
 }
 
-/// Parameters for registering a credential and storing encrypted private data.
+/// Parameters for registering a credential
 #[derive(Serialize, SchemaType, Clone, Debug)]
 pub struct RegisterCredentialParam {
     /// Public credential data.
     credential_info: CredentialInfo,
-    /// Auxiliary data. In this contract it is used to pass encrypted private
-    /// credential data to the credential storage contract.
+    /// Any additional data required by the issuer in the registration process.
     #[concordium(size_length = 2)]
     auxiliary_data:  Vec<u8>,
```

Comment:
If we keep it, do we want to store it in the `state` or do we want to emit it as an `event`? Currently, it is only an input parameter but not further processed by the smart contract. Meaning do we intend to recover the `auxiliary_data` from just the input parameter send to the smart contract?

Reply:
True, in this contract it's not used. We might keep it in the standard, but remove it from the input parameters. But in this case, would the contract be compliant with the standard?

Reply:
If we keep it in the standard and `auxiliary_data` is defined in the standard as a required part of the input parameter. Then, maybe just add a comment in this smart contract to explain the situation so people are not confused why this value is not used. 

 E.g. The CIS4 standard requires an `auxiliary_data` field. But since the issuer does not want/have extra data in our use case this input field will be empty in our example and the smart contract does not make use of the `auxiliary_data` fild.

Reply:
Yes, I'll do that. Let's keep`auxiliary_data` in the standard to allow for flexible issuer authentication. Since the issuer is identified by a public key now, it might be possible to sponsor the issuer's transactions (the issuer might not have an account even).
In the standard, we leave it to the issuer to decide how the authentication works. In this example, we keep authentication by the issuer's account.

Reply:
I'll also add `auxiliary_data` to other entrypoints that require authenticating the issuer.